lang/cs/Org.Apache.REEF.IO/FileSystem/AzureDataLake/AzureDataLakeFileSystem.cs

```diff
@@ -110,10 +110,17 @@ public void Copy(Uri sourceUri, Uri destinationUri)
         /// <exception cref="AdlsException">If copy process encounters any exceptions</exception>
         public void CopyToLocal(Uri remoteFileUri, string localName)
         {
-            TransferStatus status = _adlsClient.BulkDownload(remoteFileUri.AbsolutePath, localName);
-            if (status.EntriesFailed.Count != 0)
+            try
+            {
+                TransferStatus status = _adlsClient.BulkDownload(remoteFileUri.AbsolutePath, localName); // throws KeyNotFoundException
+                if (status.EntriesFailed.Count != 0)
+                {
+                    throw new Exception($"{status.EntriesFailed.Count} entries did not get transferred correctly");
+                }
+            }
+            catch (Exception ex)
             {
-                throw new AdlsException($"{status.EntriesFailed.Count} entries did not get transferred correctly");
+                throw new AdlsException(ex.Message);
```

Comment:
If possible always propagate the full exception otherwise you lose the stack trace.
```csharp
throw new AdlsException(ex);
```

Reply:
I need to throw the AdlsException as it is derived from IOException and that is what is mentioned as the thrown Exception for the api.